### PR TITLE
Separate PG notifier configuration

### PR DESCRIPTION
**Pull request by AkhileshK8y**
* Created a separate config for PG Notifier DB.
* Added env var for taking PG Notifier configuration.

**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by snarkychef — CHANGES_REQUESTED** [review]

File: warehouse/warehouse.go [hunk 1]
```diff
@@ -923,14 +924,21 @@ func healthHandler(w http.ResponseWriter, r *http.Request) {
...
 		host, port, user, password, dbname, sslmode)
 }
 
+func getPGNotifierConnectionString() string {
+	if jobsdb.CheckForPGNotifierEnvVars() {
+		return jobsdb.GetPGNotifierConnectionString()
+	}
+	return getConnectionString()
+}
+
```
[the remark this answers is not in the document]
Comment (AkhileshK8y) [reply warehouse/warehouse.go:941]: Done.

File: config/config.go [hunk 2]
```diff
@@ -148,6 +150,17 @@ func IsSet(key string) bool {
 	return viper.IsSet(key)
 }
 
+// IsEnvSet checks if an environment variable is set
+func IsEnvSet(key string) bool {
+	fmt.Println("Key: ", key)
```
[the remark this answers is not in the document]
Comment (AkhileshK8y) [reply config/config.go]: Done. But didn't pull IsEnvSet out of config module as there are already some other similar functions present here.

File: router/batchrouter/batchrouter.go [hunk 3]
```diff
@@ -845,7 +845,7 @@ func IsWarehouseDestination(destType string) bool {
 }
 
 func getWarehouseURL() (url string) {
-	if warehouseMode == config.EmbeddedMode {
+	if warehouseMode == config.EmbeddedMode || warehouseMode == config.PooledWHSlaveMode {
```
[the remark this answers is not in the document]
Comment (AkhileshK8y) [reply router/batchrouter/batchrouter.go]: Done.

File: warehouse/warehouse.go [hunk 4]
```diff
@@ -77,10 +77,11 @@ var (
...
 const (
-	MasterMode      = "master"
-	SlaveMode       = "slave"
-	MasterSlaveMode = "master_and_slave"
-	EmbeddedMode    = "embedded"
+	MasterMode        = "master"
+	SlaveMode         = "slave"
+	MasterSlaveMode   = "master_and_slave"
+	EmbeddedMode      = "embedded"
+	PooledWHSlaveMode = "pooled_wh_slave"
```
[the remark this answers is not in the document]
Comment (AkhileshK8y) [reply warehouse/warehouse.go:84]: Removing these redundant constants  is actually a big task and prone to break. Let us do it later.

File: warehouse/warehouse.go [hunk 5]
```diff
@@ -917,7 +918,7 @@ func healthHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func getConnectionString() string {
-	if warehouseMode == config.EmbeddedMode {
+	if warehouseMode == config.EmbeddedMode || warehouseMode == config.PooledWHSlaveMode {
```
Comment (snarkychef) [inline warehouse/warehouse.go:921]: Let's change the check here to see if warehouse db env variables are present or not, similar to what you are doing with pgnotifier.

File: warehouse/warehouse.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (snarkychef) [reply warehouse/warehouse.go:83]: Sure

File: warehouse/warehouse.go [hunk 6]
```diff
@@ -76,10 +76,11 @@ var (
...
 const (
-	MasterMode      = "master"
-	SlaveMode       = "slave"
-	MasterSlaveMode = "master_and_slave"
-	EmbeddedMode    = "embedded"
+	MasterMode        = "master"
+	SlaveMode         = "slave"
+	MasterSlaveMode   = "master_and_slave"
+	EmbeddedMode      = "embedded"
+	PooledWHSlaveMode = "pooled_wh_slave"
```
Comment (psrikanth88) [inline warehouse/warehouse.go:83]: @AkhileshK8y Are we always going to run wh master on core server when using pooled slaves (core+pooled_slaves). Shouldn't we always support running it as core+master+pooled_slaves

File: warehouse/warehouse.go (hunk 5 above)
Comment (AkhileshK8y) [reply warehouse/warehouse.go]: Done..

File: warehouse/warehouse.go (hunk 6 above)
Comment (snarkychef) [reply warehouse/warehouse.go:83]: core+master+pooled_slaves <- this is embedded mode only right? 

In multi-tenant version, we will be running slaves as separate service and warehouse master as part of backend. 

@AkhileshK8y Let's call this new mode `embedded_master` ?

**Review by snarkychef — APPROVED** [review]

File: warehouse/warehouse.go (hunk 6 above)
Comment (AkhileshK8y) [reply warehouse/warehouse.go:83]: @psrikanth88 I assume what you mean is why not separate master when we are having separate pool... Yes for now we will always run wh master inside core server.. which @snarkychef  is calling embedded_master mode..

Comment (psrikanth88) [reply warehouse/warehouse.go]: @snarkychef @AkhileshK8y  yeah, I meant separate master with slaves pooled. Cool, if we are not going to use it like that its fine.

**Review by psrikanth88 — APPROVED** [review]

**Pull request merged**